sklearn/decomposition/tests/test_dict_learning.py

```diff
@@ -238,3 +238,13 @@ def test_sparse_coder_estimator():
                        transform_alpha=0.001).transform(X)
     assert_true(not np.all(code == 0))
     assert_less(np.sqrt(np.sum((np.dot(code, V) - X) ** 2)), 0.1)
+
+
+def test_sparse_coder_mmap():
+    # Test that SparseCoder does not error by passing reading only
+    # arrays to child processes
+
+    init_dict = np.random.rand(500, 64)
```

Comment:
You should not use the global random number generator in the test. Always create a local one using:
<pre>
rng = np.random.RandomState(0)
</pre>
and then draw samples from this random number generator.